store/tikv/2pc.go

```diff
@@ -147,18 +169,20 @@ func newTwoPhaseCommitter(txn *tikvTxn) (*twoPhaseCommitter, error) {
 		}
 	}
 
-	return &twoPhaseCommitter{
-		store:     txn.store,
-		txn:       txn,
-		startTS:   txn.StartTS(),
-		keys:      keys,
-		mutations: mutations,
-		lockTTL:   lockTTL,
-	}, nil
-}
-
-func (c *twoPhaseCommitter) primary() []byte {
-	return c.keys[0]
+	// Increase lockTTL by the transaction's read time.
+	// When resolving a lock, we compare current ts and startTS+lockTTL to decide whether to clean up. If a txn
+	// takes a long time to read, increasing its TTL will help to prevent it from been aborted soon after prewrite.
+	if localElapse := time.Since(txn.startTime); localElapse < time.Second*10 {
+		return lockTTL + uint64(localElapse/time.Millisecond), nil
+	}
+	// If a long value is calculated from the local clock, there may be clock drift, take a new TS from PD to get
+	// the exact value.
+	currentTS, err := txn.store.getTimestampWithRetry(NewBackoffer(tsoMaxBackoff, context.Background()))
```

Comment:
The answer is yes for the special case that a Reader conflicts with a slow Writer, the Reader's latency will be increased by this patch. While the pay back is Writer will have greater possibilities not to be aborted+retry, thereby contributing to reducing the overall system load.